lib/cli/kit/util.rb

```diff
@@ -0,0 +1,187 @@
+        #
+        # Permission is hereby granted, free of charge, to any person obtaining
+        # a copy of this software and associated documentation files (the
+        # "Software"), to deal in the Software without restriction, including
+        # without limitation the rights to use, copy, modify, merge, publish,
+        # distribute, sublicense, and/or sell copies of the Software, and to
+        # permit persons to whom the Software is furnished to do so, subject to
+        # the following conditions:
+        #
+        # The above copyright notice and this permission notice shall be
+        # included in all copies or substantial portions of the Software.
+        #
+        # THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+        # EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+        # MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+        # NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+        # LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+        # OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+        # WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+        #
+        # Strips indentation by removing the amount of leading whitespace in the least indented
+        # non-empty line in the whole string
+        #
+        def strip_heredoc(str)
+          str.gsub(/^#{str.scan(/^[ \t]*(?=\S)/).min}/, "".freeze)
+        end
+
+        # Joins an array with commas and "and", using the Oxford comma.
+        def english_join(array)
+          return "" if array.nil?
+          return array.join(" and ") if array.length < 3
+
+          "#{array[0..-2].join(', ')}, and #{array[-1]}"
+        end
+
+        # Execute a block within the context of a variable enviroment
+        #
+        def with_environment(environment, value)
+          return yield unless environment
+
+          old_env = ENV[environment]
+          begin
+            ENV[environment] = value
+            yield
+          ensure
+            old_env ? ENV[environment] = old_env : ENV.delete(environment)
+          end
+        end
+
+        # Converts an integer representing bytes into a human readable format
+        #
+        def to_filesize(bytes, precision: 2, space: false)
+          to_si_scale(bytes, 'B', precision: precision, space: space, factor: 1024)
+        end
+
+        # Converts a number to a human readable format on the SI scale
+        #
+        def to_si_scale(number, unit = '', factor: 1000, precision: 2, space: false)
+          small_scale = %w(m µ n p f a z y)
+          big_scale = %w(k M G T P E Z Y)
+          negative = number < 0
+          number = number.abs.to_f
+
+          if number == 0 || number.between?(1, factor)
+            prefix = ""
+            scale = 0
+          else
+            scale = Math.log(number, factor).floor
```

Comment:
We can follow up in another PR, but I would highly recommend rewriting this to not use `Math.log`:

```
irb(main):004:0> Math.log(1000, 10).floor
=> 2
```

😭

Reply:
I actually rewrote this function completely during this PR. Why not use log? It seems more proper than a loop dividing the number over and over.

Reply:
Factor of 10 is not expected, it's only 1000 or 1024. I guess I could raise if that's the case.

Reply:
As long as both factors are correct for all of the prefixes we have (when the result we'd expect should be exactly `1 prefix+unit`), that can work too 👍

Reply:
Done.